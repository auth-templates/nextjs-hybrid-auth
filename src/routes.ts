export const PrivateRoutes = {
    dashboard: '/dashboard',
    account: '/account',
    leaves: '/leaves'
};

export const PublicRoutes = {
    login: '/login',
    register: '/register',
    recoverPassword: '/recover-password',
    resendConfirmationEmail: '/resend-confirmation-email',
    resetPassword: '/reset-password/:token',
    verifyAccount: '/verify-account/:token',
    twoStep: '/two-step',
}