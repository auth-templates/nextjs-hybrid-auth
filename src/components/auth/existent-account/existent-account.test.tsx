import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ExistentAccount from './existent-account';
import { PublicRoutes } from '../../routes';

describe("ExistentAccount", () => {
    it('should contain all elements', async () => {
        render(
            <MemoryRouter>
                <ExistentAccount email={'account@mail.com'} />
            </MemoryRouter>
        );

        expect(screen.getByText(/An account with the email/)).toBeInTheDocument(); 
        expect(screen.getByText(/account@mail.com/)).toBeInTheDocument();
        expect(screen.getByText(/already exists. Please go to/)).toBeInTheDocument();
        expect(screen.getByRole('link', {name: 'Sign in'})).toHaveAttribute('href', PublicRoutes.login);
        expect(screen.getByText(/page to sign in or go to/)).toBeInTheDocument();
        expect(screen.getByRole('link', {name: 'Forgot password'})).toHaveAttribute('href', PublicRoutes.recoverPassword);
        expect(screen.getByText(/page to reset your password./)).toBeInTheDocument();
    });

    it("renders correctly", () => {
        const tree = renderer.create(
            <MemoryRouter>
                <ExistentAccount email={'account@mail.com'} /> 
            </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});