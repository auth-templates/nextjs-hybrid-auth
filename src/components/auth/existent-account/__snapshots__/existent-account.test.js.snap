// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExistentAccount renders correctly 1`] = `
<div
  className="existentAccount"
>
  An account with the email 
  <div
    className="email"
  >
    account@mail.com
  </div>
   already exists. Please go to
  <a
    href="/login"
    onClick={[Function]}
  >
     Sign in 
  </a>
  page to sign in or go to
  <a
    href="/recover-password"
    onClick={[Function]}
  >
     Forgot password 
  </a>
  page to reset your password.
</div>
`;
